addons/calendar/wizard/calendar_popover_delete_wizard.py

```diff
@@ -0,0 +1,23 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import fields, models
+
+
+class CalendarProviderConfig(models.TransientModel):
+    _name = 'calendar.popover.delete.wizard'
+    _description = 'Calendar Popover Delete Wizard'
+
+
+    record = fields.Many2one('calendar.event', 'Calendar Event')
+    delete = fields.Selection([('one', 'Delete this event'), ('next', 'Delete this and following events'), ('all', 'Delete all the events')])
+
+    def close(self):
+        if not self.record or not self.delete:
+            pass
+        elif self.delete == 'one':
+            self.record.unlink()
+        elif self.delete == 'next':
+            (self.record + self.record._break_recurrence(future=True)).unlink()
+        elif self.delete == 'all':
+            self.record.recurrence_id._unlink_events()
```

Comment:
I prefer using a dictionary mapping the option to the function instead of having a bunch of if/elif

```suggestion
        if not self.record or not self.delete:
            pass
        actions = {
            'one': lambda record: record.unlink(),
            'next': lambda record: (record + record._break_recurrence(future=True)).unlink(),
            'all': lambda record: record.recurrence_id._unlink_events()
        }
        actions.get(self.delete)(self.record)
```

Reply:
That's you hacky-js brain speaking (but honestly I feel that the dictionarry mapping is less clear than if/elif as it has to use lambda functions